Target taps

olferuk opened the pull request:
I added `.targetTap` and `.targetLongPress` events. This events supplement `.tap` and `.longPress` events, but also they hold location in view where the event was fired.

So, `.targetTap` works as follows:

``` swift
hoursContainer.rx_gesture(.TargetTap(.Anywhere))
    .subscribeNext { gesture in
        switch gesture {
        case .TargetTap(let data):
            print(data.location) // prints location of the tap in hoursContainer view
            break
        default: break
        }
    }
    .addDisposableTo(bag)
```

As for `.targetLongPress`, there are `.Any`, `.Began`, `.Changed`, `.Ended` states as they are in `Pan` or `Rotate` events.

``` swift
hoursContainer.rx_gesture(.TargetLongPress(.Began), .TargetLongPress(.Ended))
    .subscribeNext { (gesture) in
        switch gesture {
        case .TargetLongPress(let data):
            print(data.location) // prints coordinates of location where user started to hold his finger and finally released
            break
        default:
            break
        }
    }
    .addDisposableTo(bag)
```

I think this will be useful when user creates a custom view with specific logic.

Commit:
Added target tap and long press events

Commit:
Added target long press states

Commit:
Bugfix

olferuk committed:
Update README.md

icanzilb commented:
Code looks good at first sight, I'm only not completely sure about the naming ... If I see the code out of the context of this PR I would not be sure what a `TargetLongPress` is. Have you considered any alternatives?

icanzilb commented:
Okay, now I'm confused what those two recognizers are supposed to do. Could you please explain what would the two code samples included in the PR description do?

Commit:
Some bugs fixed; Readme changes; -Target were renamed to -Located.

olferuk commented:
@icanzilb Ok, sure. I renamed `Target-` to `Located-`, because it is all about _where_ the gesture was performed in the view's local coordinate system.
And take a look at README, I added a few more comments about how to comprehend callback data.

icanzilb commented:
😀 ok, so the two gestures are all about the location of the tap/press? And the only piece of data the gesture provides is the tap location? 

Then what do you think of calling the gesture `TapLocation` and `LongPressLocation`? I think that would be clear to everyone then. 

What do you think?

olferuk commented:
Yes, all about location, yes, the only data, this PR is not an overwhelming improvement. :)

Well, maybe the `LocatedTap` is not the best name, but it says that it is a tap. And `LocatedLongPress` says it is a long press _gesture_. `TapLocation` looks like it is location actually, when really it is a gesture, so this can be misleading, I suppose.

icanzilb commented:
I'm not saying the contribution is not significant, I'm just trying to make sure I understand what it does. I think it should be okay to merge - I'm still hesitant about the naming but you're right that otherwise is also not very clear.

olferuk commented:
After a while I still haven't figured out the best name for this 😃

gkuhlmann14 commented:
Hey guys, what's going on with this? Working on a Rx Camera app where this could be really helpful for tap to focus.

icanzilb commented:
This PR has conflicts - it can't be merged at this time

jegnux commented:
The new [`1.0.0`](https://github.com/RxSwiftCommunity/RxGesture/releases/tag/1.0.0) release introduces a new `asLocation(in:)` operator that suits your needs. Check it out !
